src/lib/select/select.scss

```diff
@@ -45,6 +45,11 @@ $mat-select-placeholder-arrow-space: 2 * ($mat-select-arrow-size + $mat-select-a
 .mat-select-arrow-wrapper {
   display: table-cell;
   vertical-align: middle;
+
+  // When used in a box variant form-field the arrow should be centered in the box.
+  .mat-form-field-variant-box & {
+    transform: translateY(-50%);
```

Comment:
Transforming by a percentage value like this can lead to blurriness due to subpixel rendering. Generally it's better to center via flex styles